pika/connection.py

```diff
@@ -947,66 +906,60 @@ def _set_url_socket_timeout(self, value):
         self.socket_timeout = socket_timeout
 
     def _set_url_ssl_options(self, value):
-        """Deserialize and apply the corresponding query string arg"""
-        self.ssl_options = ast.literal_eval(value)
+        """Deserialize and apply the corresponding query string arg
+
+        """
+        options = ast.literal_eval(value)
+        if options is None:
+            if self.ssl_options is not None:
+                raise ValueError(
+                    'Specified ssl_options=None URL arg is inconsistent with '
+                    'the specified https URL scheme.')
+        else:
+            # Convert options to pika.SSLOptions via ssl.SSLSocket()
+            sock = socket.socket()
+            try:
+                ssl_sock = ssl.SSLSocket(sock=sock, **options)
```

Comment:
`ssl.SSLSocket` supports the important `server_hostname` arg that `ssl.wrap_socket` doesn't.